Sources/ReleaseServer/ReleaseServer.WebApi/Repositories/ReleaseArtifactRepository.cs

```diff
@@ -200,6 +199,53 @@ public List<string> GetPlatforms(string productName, string version)
 
             return platforms.OrderBy(p => p).ToList();
         }
+        
+        public BackupInformationModel RunBackup()
+        {
+            var timeStamp = DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss");
+            
+            string backupFileName = "backup_" + timeStamp + ".zip";
+            string backupArchiveFileName = Path.Combine(BackupRoot, backupFileName); 
+            
+            //Clear the backup folder first
+            DeleteFilesInPath(BackupRoot);
+            
+            //Create the backup -> zip the whole ArtifactRoot folder
+            ZipFile.CreateFromDirectory(ArtifactRoot, backupArchiveFileName);
+
+            return new BackupInformationModel
+            {
+                FullPath = backupArchiveFileName,
+                FileName = backupFileName
+            };
+        }
+
+        public void RestoreBackup(ZipArchive backupPayload)
+        {
+            try
+            {
+                //Clear the whole artifact root directory
+                Directory.Delete(ArtifactRoot, true);
+                Directory.CreateDirectory(ArtifactRoot);
+                
+                backupPayload.ExtractToDirectory(ArtifactRoot);
+            }
+            catch (Exception e)
+            {
+                Logger.LogCritical("unexpected error during restoring the backup");
+                throw;
+            }
+        }
+
+        private void DeleteFilesInPath(string path)
+        {
+            var directoryInfo = new DirectoryInfo(path);
+
+            foreach (var file in directoryInfo.GetFiles())
+            {
+                file.Delete();
+            }
+        }
```

Comment:
As noted in the comment on line 211 method should be replaced by the extension method `DeleteFiles()` on `DirectoryInfo`.

Reply:
Realized, see above.